### PR TITLE
Customer

### PR DESCRIPTION
**Pull request by adeeb27**
User can select the different categories on the Customer Landing Page to view specific Menu items related to a tag. E.g. Appetizer, Main Course, etc. The back button (cross on the top) of the menu page is not working as of now. The category name and the menu items inside (boxes) generate dynamically, i.e. they are received and from the backend and displayed.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <component name="ProjectModuleManager">
     <modules>
       <module fileurl="file://$PROJECT_DIR$/potatopeeps.iml" filepath="$PROJECT_DIR$/potatopeeps.iml" />
+      <module fileurl="file://$PROJECT_DIR$/potatopeeps.iml" filepath="$PROJECT_DIR$/potatopeeps.iml" />
```
Comment (Gabetn) [inline .idea/modules.xml:6]: What is the purpose of this? @adeeb27 
Seems to be a duplicate of line 5

**Review by Gabetn — CHANGES_REQUESTED** [review]
.DS_Store files should be removed. 
I am in the progress of doing this.

customerLanding.css should not have been copied in its entirety, rather it should have been merged from the UI branch

File: pom.xml [hunk 2]
```diff
@@ -107,4 +107,4 @@
 		</plugins>
 	</build>
 
-</project>
+</project>
```
Comment (Gabetn) [inline pom.xml:110]: What was the purpose of these edits in pom.xml @dibbor ?
It looks like nothing was changed, just deleted then re-added

**Review by Gabetn — CHANGES_REQUESTED** [review]

File: .idea/modules.xml (hunk 1 above)
Comment (adeeb27) [reply .idea/modules.xml:6]: IntelliJ probably autogenerated this, didn't type it in manually.

**Review by EBruchet — CHANGES_REQUESTED** [review]
**Issues/Errors Found in Initial PR:**
- Stylesheets included in HTML format instead of React JS imports (**RESOLVED**)
- Other stylesheets are imported unnecessarily including `style.css`, `manager.css`
  - Elected not to resolve as of now, as we still need to see what the UI team will push in terms of updated CSS. On the other hand, `style.css` and `manager.css` have no place in `Customer.js` and shouldn't be imported there regardless, but as said before - ensure the `.css` files edit only apply to the pages they are named after.
- `HandleClick()` functionality was placed inside of the top-level `Customer` component instead of a lower-level component to which it is more closely related (**RESOLVED**)
  - Note, not only was this shifted down but the `HandleClick` functionality was changed, as the previous implementation was hard to understand and took a long time to load the MenuItems into the window.
- Inline styling found across the `Customer.js` file, and as mentioned before - this should be **AVOIDED**, for the sake of clarity and consistency. `.css` files should contain all the styling for their respective pages.
- `CustomerMenuItemList` from `MenuItem.js` was not used (**RESOLVED**)
- Unnecessary resetting of state found on line `48` (**RESOLVED**)
- `if` statement for selecting MenuItem tag type should be replaced with use of Router History to navigate in between pages (**RESOLVED**)
- `X` button on tag-specific menu windows failed to send user back to the Customer page (**RESOLVED**)


**Further Changes:**
- `FilterMenuItemList` was modified to work with the `Customer.js` file, but this modification broke the corresponding filter functionality in `Manager.js`. Will branch off of this commit and fix accordingly.
- Various styling errors on the displayed MenuItem windows in Customer view, should be resolved by the UI team.
- While all tags are an option on the Customer page, requires scrolling across to view the rest - another fix to be resolved by the UI team.

File: pom.xml (hunk 2 above)
Comment (dibbor) [reply pom.xml:110]: I was having problems building it initially so I googled and tried stuff but later just copy pasted the original one.

**Comment by Gabetn** [comment]
### Merges
I have merged the changes to filtering from Customer-FilterMenuItem, as well as dev which contained the Login updates. 
In the merges I modified the redirects within `Customer.js` such that the **TableNumberSelect** is displayed first. 
_After_ selecting a table number the **CustomerLanding** page is pushed onto the history along side **_tableNum_** to keep track of the tableNumber (will come in handy when dealing with DiningSessions inside Customer. _Note_: This required the routing for CustomerLanding inside `App.js`

Routing could have been simplified by the use of a if-else around the render function on Customer to allow conditional rendering based on whether a tablenumber was selected. (If not render _TableNumberSelect_, if so render _CustomerLandingPage_). However my current approach seems more in-line with the current implementations of routing. 

**TLDR**: This pull request should now, aside from some cleaning, be ready to merged into `dev`. Also could use a second opinion on how I implemented the routing.

**Review by Gabetn — APPROVED** [review]

**Review by EBruchet — APPROVED** [review]
